### PR TITLE
[add] OpenAIRE publication engine

### PR DESCRIPTION
**Pull request by jibe-b**
and modifies OpenAIRE datasets engine.

Just a short PR to build on #1075

### CODE REVIEW

File: searx/settings.yml [hunk 1]
```diff
@@ -424,7 +424,18 @@ engines:
     title_query : metadata/oaf:entity/oaf:result/title/$
     content_query : metadata/oaf:entity/oaf:result/description/$
     categories : science
-    shortcut : oa
+    shortcut : oad
+    timeout: 5.0
+
+  - name : openairepublications
```
Comment (asciimoo) [inline searx/settings.yml:430]: Add `engine: json_engine` to specify the engine you want to use

**Review by asciimoo — APPROVED** [review]
Great, thanks

**Comment by jibe-b** [comment]
Well, openairepublications and openairedatasets fail

>  File "/searx/searx/engines/json_engine.py", line 102, in response
    url = query(result, url_query)[0]
IndexError: list index out of range
ERROR:searx.search:engine openairepublications : exception : list index out of range


@asciimoo Any hint on where the configuration of the json_engine would fail?

**Comment by asciimoo** [comment]
@jibe-b i've added some error handling to the json engine, it should work now.

**Comment by jibe-b** [comment]
Thanks for doing this quickly but it still fails (indeed after pulling to last commit 181f1c6)

>ERROR:searx.search:engine openairepublications : exception : list index out of range
Traceback (most recent call last):
  File "/searx/searx/search.py", line 118, in search_one_request_safe
    search_results = search_one_request(engine, query, request_params, start_time, timeout_limit)
  File "/searx/searx/search.py", line 110, in search_one_request
    return engine.response(response)
  File "/searx/searx/engines/json_engine.py", line 105, in response
    url = query(result, url_query)[0]
IndexError: list index out of range

**Comment by asciimoo** [comment]
@jibe-b can you test it now (7a9b18e9)? Seems there are results with no url.

**Comment by jibe-b** [comment]
@asciimoo it's working now

image

But quite slow…

image

(5sec timeout was not enough)

**Pull request merged**